RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -3293,6 +3299,33 @@ PFAlgo::reconstructCluster(const reco::PFCluster& cluster,
 
 }
 
+void
+PFAlgo::setHcalDepthInfo(reco::PFCandidate & cand, const reco::PFCluster& cluster) const {
+    std::array<double,7> energyPerDepth; 
+    std::fill(energyPerDepth.begin(), energyPerDepth.end(), 0.0);
+    for (auto & hitRefAndFrac : cluster.recHitFractions()) {
+        const auto & hit = *hitRefAndFrac.recHitRef();
+        if (DetId(hit.detId()).det() == DetId::Hcal) {
+            if (hit.depth() == 0) {
+                edm::LogWarning("setHcalDepthInfo") << "Depth zero found";
+                continue;
+            }
+            if (hit.depth() < 1 || hit.depth() > 7) {
+                throw cms::Exception("CorruptData") << "Bogus depth " << hit.depth() << " at detid " << hit.detId() << "\n";
+            }
+            energyPerDepth[hit.depth()-1] += hitRefAndFrac.fraction()*hit.energy();
+        }
+    }
+    double sum = std::accumulate(energyPerDepth.begin(), energyPerDepth.end(), 0.);
+    std::array<float,7> depthFractions;
+    std::fill(depthFractions.begin(), depthFractions.end(), 0.f);
```

Comment:
This line could more conveniently move with an ```else``` after ```if (sum > 0) {...}```

Reply:
done